### PR TITLE
changing BuildSpecSchema to support new yaml

### PR DESCRIPTION
**Pull request by joker2411**
## Description of the change

> The required yaml structure is changed. This is to support that. Ticket [Link](https://linear.app/rudderstack/issue/PRML-784/update-model-with-the-new-yaml).

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/predictions/profiles_mlcorelib/py_native/attribution_report.py [hunk 1]
```diff
@@ -26,38 +29,54 @@ class AttributionModel(BaseModelType):
     BuildSpecSchema = {
         "type": "object",
         "properties": {
-            **EntityKeyBuildSpecSchema["properties"],
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/py_native/attribution_report.py:29]: I think this is default `entity_key` for a model right? Why remove this?

Comment (dpatchigolla) [reply src/predictions/profiles_mlcorelib/py_native/attribution_report.py:29]: Shouldn't we have this in place of conversion/properties/Entity_key? Can you check with Shashank if that's feasible to define `EntityKeyBuildSpecSchema` in a nested object? Else any disadvantage in not using this? 
There's a notion of model entity, which lets us define user.Var() in the spec where user is the entity of the model. I think that may be lost if we don't use it.

File: src/predictions/profiles_mlcorelib/py_native/attribution_report.py [hunk 2]
```diff
@@ -26,38 +29,54 @@ class AttributionModel(BaseModelType):
...
+                            "required": ["from", "timestamp"],
+                        },
+                    },
+                    CONVERSION_VARS: {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "name": {"type": "string"},
+                                "timestamp": {"type": "string"},
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/py_native/attribution_report.py:53]: Are `user.Var("first_paid_date")` etc counted as strings?

**Review by dpatchigolla — COMMENTED** [review]
Can you address the comments? Also can you update the wyze repo too and create a parallel pr with this new structure after you run it locally?

File: src/predictions/profiles_mlcorelib/py_native/attribution_report.py (hunk 2 above)
Comment (joker2411) [reply src/predictions/profiles_mlcorelib/py_native/attribution_report.py:53]: yes.

File: src/predictions/profiles_mlcorelib/py_native/attribution_report.py (hunk 1 above)
Comment (joker2411) [reply src/predictions/profiles_mlcorelib/py_native/attribution_report.py:29]: checked with Shashank. we need to define `entity_key` in model_spec directly to use user.Var() syntax. 
Moving the entity_key to model_spec

**Review by dpatchigolla — APPROVED** [review]

**Pull request merged**